### PR TITLE
90% - Lig 405 add optional description

### PR DESCRIPTION
**Pull request by markwallsgrove**
Adding the ability to append meta data to each log statement for filtering / documentation

### CODE REVIEW

File: js/logging.js [hunk 1]
```diff
@@ -91,10 +91,10 @@ loggingModule.factory(
     "applicationLoggingService",
     ["$log","$window", "LOGGING_CONFIG", function($log, $window, LOGGING_CONFIG){
         return({
-            error: function(message){
+            error: function(message, desc){
                 if (LOGGING_CONFIG.LOGGING_TYPE !== 'none') {
                   // preserve default behaviour
-                  $log.error.apply($log, arguments);
```
Comment (russellhill) [inline js/logging.js]: I'm confused about where "arguments" came from originally...

Comment (markwallsgrove) [inline js/logging.js]: https://docs.nodejitsu.com/articles/javascript-conventions/what-is-the-arguments-object - magic

Comment (russellhill) [inline js/logging.js]: heh...interesting...

Comment (russellhill) [inline js/logging.js]: are we not therefore losing this arguments information now?

Comment (markwallsgrove) [inline js/logging.js]: Beforehand the call was basically doing `console.error([message])`. When I then appended `desc`, the output was  `console.error([message, desc])`, which looked weird. I believe we only want to show the message on the console.

Comment (russellhill) [inline js/logging.js]: ok

File: js/logging.js [hunk 2]
```diff
@@ -127,7 +128,8 @@ loggingModule.factory(
                         data: angular.toJson({
                             url: $window.location.href,
                             message: message,
-                            type: "debug"
+                            type: "debug",
+                            desc: desc
```
Comment (russellhill) [inline js/logging.js]: do we need to check if desc has been passed here?

Comment (markwallsgrove) [inline js/logging.js]: it will default to `null`.

Comment (markwallsgrove) [inline js/logging.js]: Actually `null` would be too logical for JS. Javascript removes the key/value if the value is undefined.
```
x = function(dev) { return JSON.stringify({'a' : 'b', 'blah' : dev}); }
function (dev) { return JSON.stringify({'a' : 'b', 'blah' : dev}); }
x()
"{"a":"b"}"
```

**Pull request merged**